RevenueCatUI/Templates/Components/Packages/PackageGroup/PackageGroupComponentViewModel.swift

```diff
@@ -36,20 +36,42 @@ class PackageGroupComponentViewModel {
         let info = try Self.getPackages(component: component, offering: offering)
         self.defaultPackage = info.defaultPackage
 
-        self.stackComponentViewModel = try self.component.toStackComponentViewModel(
-            components: info.availablePackageComponents.map { .package($0) },
-            localizedStrings: localizedStrings,
-            offering: offering
+        let componentViewModels = try info.availablePackageComponents.map { info in
+            try PackageComponentViewModel(
+                localizedStrings: localizedStrings,
+                component: info.component,
+                offering: offering,
+                package: info.package
+            ).stackViewModel
+        }.map(PaywallComponentViewModel.stack)
+
+        self.stackViewModel = StackComponentViewModel(
+            component: .init(
+                components: [], // Empty on purpose because we are feeding this view models already
+                dimension: component.stack.dimension,
+                width: component.stack.width,
+                spacing: component.stack.spacing,
+                backgroundColor: component.stack.backgroundColor,
+                padding: component.stack.padding,
+                margin: component.stack.margin,
+                cornerRadiuses: component.stack.cornerRadiuses,
+                border: component.stack.border
+            ),
+            viewModels: componentViewModels
```

Comment:
I don't love this so I _might_ try to reworked this but...

Above this, available packages are mapped into an array of `PackageComponentViewModel` and then converted to an array of stack view models.

This needs to stay as an array of view models because the `PackageComponentViewModel` holds the `Package`. However, `StackComponentViewModel`s are usually made with the raw components (and the view models are created in the constructor.

But because we already have the view models, I added another initializer to `StackComponentViewModel` that takes the component AND the view models. But (another one), this expects a `StackComponent` (and not a `PackageStackComponent` so we need to map that and also pass in the view models.

I'm okay with the rest of this PR except for this. But I wanted to explain myself anyway before I try to change it since it _does_ work 😇

Reply:
I agree that this is a bit confusing, especially this line makes me pause every time I read it haha: 
```swift
 components: [], // Empty on purpose because we are feeding this view models already
```

I know what's happening, but it's not obvious at first glance. Would be nice to improve this, but not a blocker for me.

Reply:
> I know what's happening, but it's not obvious at first glance. Would be nice to improve this, but not a blocker for me.

I'll create a cleanup task for this for me! This isn't really affecting much else and I don't want this to slow things down so I'll sleep / take the weekend on this and see if sometime magical pops into my head 😇

Reply:
That sounds like a fine plan!